ext/datasource/helper.go

```diff
@@ -151,9 +160,9 @@ func HotSpotParamRuleJsonArrayParser(src []byte) (interface{}, error) {
 	}
 
 	rules := make([]*hotspot.Rule, 0)
-	err := json.Unmarshal(src, &rules)
-	if err != nil {
-		return nil, err
+	if err := json.Unmarshal(src, &rules); err != nil {
+		desc := fmt.Sprintf("failed to convert source bytes to []*hotspot.Rule, err: %s", err.Error())
+		return nil, NewError(ConvertSourceError, desc)
 	}
```

Comment:
Maybe you should refine NewError() naming for HotSpotParamRulesUpdater

Reply:
Addressed.